### PR TITLE
Lint

### PR DESCRIPTION
**Pull request by rgarcia**
Mostly adding comments to exported types, methods, etc.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,9 +7,16 @@ build/container: build/discoverd
 build/discoverd: Godeps *.go agent/*.go
 	godep go build -o build/discoverd
 
+lint:
+	go get github.com/golang/lint
+	golint *.go **/*.go
+
+fmt:
+	gofmt -w=true *.go **/*.go
```
Comment (titanous) [inline Makefile]: No need for the `=true` and add the `-s` flag.

Comment (rgarcia) [inline Makefile]: 67e861b

**Comment by titanous** [comment]
LGTM. Squash the last commit and I'll merge.

File: Makefile [hunk 2]
```diff
@@ -7,9 +7,16 @@ build/container: build/discoverd
 build/discoverd: Godeps *.go agent/*.go
 	godep go build -o build/discoverd
 
+lint:
+	go get github.com/golang/lint
+	golint *.go **/*.go
+
+fmt:
+	gofmt -w -s *.go **/*.go
```
Comment (titanous) [inline Makefile]: Oh, just remembered that the file list can be simplified to `.`.

Comment (rgarcia) [inline Makefile]: de3082a

**Comment by titanous** [comment]
Thanks!

**Pull request merged**